pages/project-version/middleware/index.js

```diff
@@ -304,9 +304,13 @@ const getPreviousProtocols = (firstVersion, previousVersion, grantedVersion) =>
   const previous = get(previousVersion, 'data.protocols', []).filter(Boolean).filter(p => !p.deleted).map(p => p.id);
   const granted = get(grantedVersion, 'data.protocols', []).filter(Boolean).map(p => p.id);
   const showDeleted = uniq([ ...previous, ...granted ]);
-  const allSteps = get(previousVersion, 'data.protocols', []).filter(Boolean).filter(p => !p.deleted).map(p => p.steps);
+  let allSteps = get(previousVersion, 'data.protocols', []).filter(Boolean).filter(p => !p.deleted).map(p => p.steps);
```

Comment:
Reusing allStep is confusing,and theres a lot of repeated logic here. I'd suggest renaming to previous/first steps, tbh I think it's clearer if these are extracted to a function:
```suggestion
  function extractActiveSteps(version) {
    return get(version, 'data.protocols', [])
      .filter(Boolean)
      .filter(p => !p.deleted)
      .map(p => p.steps)
      .map((element) => element && Array.isArray(element) ? element.filter(s => !s.deleted) : element);
  }
  
  const steps = extractActiveSteps(previousVersion);
  const firstSteps = extractActiveSteps(firstVersion);
  const grantedSteps = extractActiveSteps(grantedVersion);
```